diesel/src/expression/mod.rs

```diff
@@ -110,20 +131,20 @@ impl<T: Expression> AsExpression<T::SqlType> for T {
 /// Columns will implement this for their table. Certain special types, like
 /// `CountStar` and `Bound` will implement this for all sources. All other
 /// expressions will inherit this from their children.
-pub trait SelectableExpression<QS>: Expression {
+pub trait SelectableExpression<QS>: AppearsOnTable<QS> {
```

Comment:
Maybe add a line about caring about nullability to the docs (esp. as this is a subtrait of `AppearsOnTable` now)? Actually, to be really explicit you can still keep `Expression` as supertrait (it's referred to in the docs).